src/components/CatchUnsavedChangesModal.tsx

```diff
@@ -0,0 +1,90 @@
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+import { Text } from "./atomic";
+import SelectOptionModal, { SelectOptionModalProps } from "./SelectOptionModal";
+
+function DefaultModalChildren() {
+  return (
+    <div>
+      <Text>Changes you made may not be saved.</Text>
+    </div>
+  );
+}
+
+interface CatchUnsavedChangesModalProps {
+  unsavedChangesExist: boolean;
+  onRouteChangePrevented?: (futureURL: string) => void;
+}
+function CatchUnsavedChangesModal({
+  unsavedChangesExist,
+  onRouteChangePrevented = () => {},
+  title = "Leave page?",
+  secondaryButtonText = "Cancel",
+  primaryButtonText = "Leave",
```

Comment:
I think cancel should be the primary option (in the browser dialogue too) since clicking "Cancel" on accident will be better than clicking "Leave" and losing all progress

Reply:
Cancel is already the primary focused button. Typically, the secondary button is the one that's focused by default because of exactly what you said.

Though this prop naming might be a little confusing? But "Primary Button" refers to the button that does the primary action.